Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -439,30 +440,38 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
                         this.writeEmitter.fire(this.endOfLine);
                         if (result === null) {
                             this.writeEmitter.fire(localize("build.run.terminated", "Build run was terminated.") + this.endOfLine);
+                            buildTerminated = true;
                             resolve(-1);
```

Comment:
Is the real problem that `resolve(-1)` doesn't exit the function and we immediately send `resolve(0)` right after this?  What happens if you just add a `return;` after line 444?  Maybe the reason this change works is because it now ignores the -1 and 0 returned in the failed build case.

(It's possible you might also need to do something to deal with multiple `resolve`s being called if both the `error` block and the `close` block can execute.)

Reply:
In new changes, I kept the result of the promise and checked for when it is -1.